### PR TITLE
pg, set valueformat of row description in bind command

### PR DESCRIPTION
**Pull request by susongyan**
when using with gorm ,   select int column,     gorm  will check  valueFormat in the row description  to decide how to deal with the column value , but  the format is always  0(text),  then gorm treat  int8 array as string, then it failed to convert 

image

image

cuz format = 0, then `strconv.ParseUint(string(src), 10, 64)`
error message as follows:

convert field 0 failed: strconv.ParseInt: parsing "\x00\x00\x00\x00\x00\x00\x00\x16": invalid syntax [437.417ms] [rows:0] INSERT INTO "t_student_info_copy" ("student_name","id") VALUES ('go name',22) RETURNING "id" &{0xc0000ee510 convert field 0 failed: strconv.ParseInt: parsing "\x00\x00\x00\x00\x00\x00\x00\x16": invalid syntax 0 0xc000242380 0}

### CODE REVIEW

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
Hi @susongyan, 
Could you add some unit tests for this change?

**Comment by susongyan** [comment]
> Hi @susongyan, Could you add some unit tests for this change?

ok,  i'll add test for portal after bind

> Hi @susongyan, Could you add some unit tests for this change?

ok ,  i'll add  data format test for  port.describe

**Comment by RaigorJiang** [comment]
Many CI actions failed, I try rerun first.

File: db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLRowDescriptionPacket.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
  * Row description packet for PostgreSQL.
  */
 @RequiredArgsConstructor
+@Getter
```
Comment (TeslaCN) [inline db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLRowDescriptionPacket.java:33]: Do we need this getter in production code? If not, please access this field by Mockito Plugins or reflection.

Comment (susongyan) [reply db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLRowDescriptionPacket.java:33]: only in test,  i'l remove it

File: proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java [hunk 2]
```diff
@@ -145,7 +153,17 @@ void assertExecuteSelectStatementAndReturnAllRows() throws SQLException {
...
+
+        PostgreSQLPacket portalDescription = portal.describe();
+        assertThat(portalDescription, instanceOf(PostgreSQLRowDescriptionPacket.class));
+        Optional<Collection<PostgreSQLColumnDescription>> columnDescriptions = ReflectionUtils.getFieldValue(portalDescription, "columnDescriptions");
+        assertTrue(columnDescriptions.isPresent());
+        Iterator<PostgreSQLColumnDescription> columnDescriptionIterator = columnDescriptions.get().iterator();
+        PostgreSQLColumnDescription textColumnDescription = columnDescriptionIterator.next();
+        PostgreSQLColumnDescription intColumnDescription = columnDescriptionIterator.next();
+        assertEquals(textColumnDescription.getDataFormat(), PostgreSQLValueFormat.TEXT.getCode());
+        assertEquals(intColumnDescription.getDataFormat(), PostgreSQLValueFormat.BINARY.getCode());
```
Comment (TeslaCN) [inline proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java:165]: Please use `assertThat` here.

File: proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java [hunk 3]
```diff
@@ -132,8 +138,10 @@ void assertGetName() throws SQLException {
     @Test
     void assertExecuteSelectStatementAndReturnAllRows() throws SQLException {
         QueryResponseHeader responseHeader = mock(QueryResponseHeader.class);
-        QueryHeader queryHeader = new QueryHeader("schema", "table", "columnLabel", "columnName", Types.INTEGER, "columnTypeName", 0, 0, false, false, false, false);
-        when(responseHeader.getQueryHeaders()).thenReturn(Collections.singletonList(queryHeader));
+        QueryHeader queryHeader = new QueryHeader("schema", "table", "columnLabel", "columnName", Types.VARCHAR, "columnTypeName", 0, 0, false, false, false, false);
+        QueryHeader intColumnQueryHeader = new QueryHeader("schema", "table", "columnLabel", "columnName", Types.INTEGER, "columnTypeName", 0, 0, false, false, false, false);
+        when(responseHeader.getQueryHeaders()).thenReturn(Arrays.asList(queryHeader, intColumnQueryHeader));
+
```
Comment (TeslaCN) [inline proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java:144]: Please remove blank line.

File: proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java [hunk 4]
```diff
@@ -145,7 +153,17 @@ void assertExecuteSelectStatementAndReturnAllRows() throws SQLException {
         List<PostgreSQLValueFormat> resultFormats = new ArrayList<>(Arrays.asList(PostgreSQLValueFormat.TEXT, PostgreSQLValueFormat.BINARY));
         Portal portal = new Portal("", preparedStatement, Collections.emptyList(), resultFormats, backendConnection);
         portal.bind();
-        assertThat(portal.describe(), instanceOf(PostgreSQLRowDescriptionPacket.class));
+
```
Comment (TeslaCN) [inline proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java:156]: Please remove blank line.

File: proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java [hunk 5]
```diff
@@ -145,7 +153,17 @@ void assertExecuteSelectStatementAndReturnAllRows() throws SQLException {
...
+        PostgreSQLPacket portalDescription = portal.describe();
+        assertThat(portalDescription, instanceOf(PostgreSQLRowDescriptionPacket.class));
+        Optional<Collection<PostgreSQLColumnDescription>> columnDescriptions = ReflectionUtils.getFieldValue(portalDescription, "columnDescriptions");
+        assertTrue(columnDescriptions.isPresent());
+        Iterator<PostgreSQLColumnDescription> columnDescriptionIterator = columnDescriptions.get().iterator();
+        PostgreSQLColumnDescription textColumnDescription = columnDescriptionIterator.next();
+        PostgreSQLColumnDescription intColumnDescription = columnDescriptionIterator.next();
+        assertEquals(textColumnDescription.getDataFormat(), PostgreSQLValueFormat.TEXT.getCode());
+        assertEquals(intColumnDescription.getDataFormat(), PostgreSQLValueFormat.BINARY.getCode());
+
```
Comment (TeslaCN) [inline proxy/frontend/type/postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/PortalTest.java:166]: Please remove blank line.

File: db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLColumnDescription.java [hunk 6]
```diff
@@ -41,12 +42,20 @@ public final class PostgreSQLColumnDescription {
     
     private final int typeModifier = -1;
     
-    private final int dataFormat = 0;
+    private int dataFormat = PostgreSQLValueFormat.TEXT.getCode();
```
Comment (TeslaCN) [inline db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLColumnDescription.java:45]: Please make this field final.

File: db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLColumnDescription.java [hunk 7]
```diff
@@ -41,12 +42,20 @@ public final class PostgreSQLColumnDescription {
...
     
     public PostgreSQLColumnDescription(final String columnName, final int columnIndex, final int columnType, final int columnLength, final String columnTypeName) {
         this.columnName = columnName;
         this.columnIndex = columnIndex;
         this.columnLength = columnLength;
         typeOID = Types.ARRAY == columnType ? PostgreSQLArrayColumnType.getTypeOid(columnTypeName) : PostgreSQLColumnType.valueOfJDBCType(columnType).getValue();
```
Comment (TeslaCN) [inline db-protocol/postgresql/src/main/java/org/apache/shardingsphere/db/protocol/postgresql/packet/command/query/PostgreSQLColumnDescription.java:51]: Consider using  `this(a, b, ...)` here.

**Review by TeslaCN — APPROVED** [review]
LGTM

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Merged, thank you! @susongyan @TeslaCN

**Pull request merged**